ramls/schema/anonymize-loans-response.json

```diff
@@ -0,0 +1,22 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "Response schema for anonymize loans request",
+  "type": "object",
+  "properties": {
+    "anonymizedLoans": {
+      "description": "Successfully anonymized loan ids",
+      "type": "array",
+      "items":{
+        "type": "string"
+      }
+    },
+    "notAnonymizedLoans": {
+      "description": "Not anonymized loan ids due to open fees and fines",
```

Comment:
There are, of course, many reasons for a loan to be not anonymized.  RIght now, it's only about fees and fined, and the response is an aggregation of all not anonymized loans for various reasons. If in the future there will be a need to distinquish between those reasons, we can add it then.
There were no direct requirements to this, but I think returning a list of not anonymized loan ids sound like a good idea.

What do you think should be returned here?

Reply:
We also discussed using https://github.com/folio-org/raml/blob/5f350c1f257d1bcf15cb54639ec452ac0e70300b/schemas/error.schema instead of reason, to keep it consistent with other APIs

Reply:
This structure is a trade off between making the API less brittle to the expectations of the reference UI (by not assuming what error conditions can occur) and by not making the client work extra hard to aggregate the un-anonymized loans by reason.

It relies on only a single reason be applicable for a loan (or that a loan could appear under multiple reasons)

Reply:
New proposed response schema:
`{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "description": "Response schema for anonymize loans request",
  "type": "object",
  "properties": {
    "anonymizedLoans": {
      "description": "Successfully anonymized loan ids",
      "type": "array",
      "items": {
        "type": "string"
      }
    },
    "notAnonymizedLoans": {
      "type": "object",
      "$ref": "raml-util/schemas/errors.schema",
      "readonly": true
    }
  },
  "additionalProperties": false
}`

Reply:
The following response schema is used:
`
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "description": "Response schema for anonymize loans request",
  "type": "object",
  "properties": {
    "anonymizedLoans": {
      "description": "Successfully anonymized loan ids",
      "type": "array",
      "items":{
        "type": "string"
      }
    },
    "errors": {
      "description": "List of errors",
      "id": "errors",
      "type": "array",
      "items": {
        "type": "object",
        "$ref": "../raml-util/schemas/error.schema"
      }
    }
  },
  "additionalProperties": false
}
`